onnxruntime/core/providers/cuda/nn/instance_norm.cc

```diff
@@ -100,10 +100,19 @@ Status InstanceNorm<T>::ComputeInternal(OpKernelContext* p_op_kernel_context) co
     CudnnTensor stats_desc;
     ORT_RETURN_IF_ERROR(stats_desc.Set(std::array<int64_t, 4>{1, stats_count, 1, 1}, CudnnTensor::GetDataType<CudaT>()));
 
-    auto mean = GetScratchBuffer<CudaT>(stats_count);
-    auto variance = GetScratchBuffer<CudaT>(stats_count);
-    auto unused_scale = GetScratchBuffer<CudaT>(stats_count);
-    auto unused_bias = GetScratchBuffer<CudaT>(stats_count);
```

Comment:
>GetScratchBuffer [](http://example.com/codeflow?start=23&length=16)

This interface is terribly wrong. It's argument names count_or_bytes. It is templated and it returns a typed smartptr, all of which is two big lies. And an invitation to pass the count of items.
It requires size in bytes and returns a bytes allocated buffer. It should never been a template.